pkg/apiserver/apiserver.go

```diff
@@ -81,22 +95,94 @@ func NewAPIServer() *cobra.Command {
 			return runE(flags, log)
 		},
 	}
+	flags.ConfigFlags.AddFlags(cmd.Flags())
 	cmd.Flags().StringVar(&flags.address, "address", "0.0.0.0:8080", "Address to bind this API server on.")
 	cmd.Flags().StringVar(&flags.TLSCertFile, "tls-cert-file", "", "File containing the default x509 Certificate for HTTPS. If not provided no TLS security shall be enabled")
 	cmd.Flags().StringVar(&flags.TLSPrivateKeyFile, "tls-private-key-file", "", "File containing the default x509 private key matching --tls-cert-file.")
 	cmd.Flags().StringArrayVar(&flags.CORSAllowedOrigins, "cors-allowed-origins", []string{"*"}, "List of allowed origins for CORS, comma separated. An allowed origin can be a regular expression to support subdomain matching. If this list is empty CORS will not be enabled.")
-	oidc.AddOIDCPFlags(&flags.OIDCOptions, cmd.Flags())
+	cmd.Flags().StringArrayVar(&flags.AuthorizationMode, "authorization-mode", []string{"OIDC"}, "Ordered list of plug-ins to do authorization on secure port. Comma-delimited list of: "+strings.Join(auth.RegisteredAuthProviders(), ","))
```

Comment:
Please use `authentication` not `authorization` in this PR

Reply:
done